### PR TITLE
#172 All/saved companies functionality

### PR DESCRIPTION
**Pull request by Lvyshnevska**
All/saved companies functionality implemented. Field "saved" added to ProfileSerializer. It returns True, if company is in saved companies list of user. User's Companies saved list  returns by passing query param by "api/profiles/" endpoint. Due to this SavedCompanies functionality and test updated a little.

### CODE REVIEW

**Review by ProDG — APPROVED** [review]

**Review by mehalyna — APPROVED** [review]

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:78]: query set in this situation can't raise ValueError, this try-except is not needed here

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:69]: please create `user_id` variable instead of `userid`

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:66]: I told you about FilterQueryParamsSerializer, so you should have something like this here:

filters = FilterQueryParamsSerializer(data=self.request.query_params.get("filters"))

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:89]:
```suggestion
        if filters.is_valid():
            if filters.get("saved"):
                queryset = queryset.filter(saved_list__user=self.request.user)
```

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:84]: Too many returns in one function, especially if you will want to use multiple filters

queryset = queryset.filter(comp_activity__name=activity_type)

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:82]: queryset = queryset.filter(comp_registered=True)

File: profiles/views.py
Comment (GeorgiyII) [inline profiles/views.py:80]: queryset = queryset.filter(comp_is_startup=True)

File: profiles/views.py
Comment (Lvyshnevska) [reply profiles/views.py:78]: But if we pass as "userid" value that not an integer, ValueError occurs. So try/except lets us to avoid this error and to return queryset. Or there is another way to solve this issue?

File: profiles/views.py
Comment (Lvyshnevska) [reply profiles/views.py]: Fixed

File: profiles/views.py
Comment (Lvyshnevska) [reply profiles/views.py]: Fixed

File: profiles/views.py
Comment (Lvyshnevska) [reply profiles/views.py]: I'll improve it next sprint

File: profiles/serializers.py
Comment (Lvyshnevska) [reply profiles/serializers.py]: I'll improve it next sprint

File: profiles/serializers.py
Comment (ProDG) [inline profiles/serializers.py:1]: Is it used? If not — remove the import.

File: profiles/serializers.py
Comment (ProDG) [inline profiles/serializers.py:35]: In case someone will decide to use this serialize in List endpoint, we'll have a trouble. 

For _each_ of the profile, this code (containing call to DB) will be executed, so there will be N+1 problem.

If we're sure that the serializer will be used only for details endpoint (and in this case adding `Details` word into the name would be helpful), we can keep it as is.

If not (let's say, we know for sure that it will be used for List endpoint), we can do the next trick:

1. Retrieve PKs of all comanies, saved by current user.
2. Put this data as a list (or, better — frozenset) into serializer context (by overriding `get_serializer_context()`)
3. This method will contain a single string of code, something like `return obj.pk in self.context['saved_companies_pks']`
4. .. and also, you can combine these approaches — check if there is `saved_companies_pks` in context, and do the next steps based on the result.

File: profiles/serializers.py
Comment (ProDG) [inline profiles/serializers.py:95]: I have strong feeling that something is very wrong here. 

Serializers have nothing to do with query parameters.

File: profiles/views.py
Comment (ProDG) [inline profiles/views.py:63]: Nope. It _must_ be implemented using this feature of DRF:

https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend

Comment (ProDG) [reply profiles/views.py:63]: You can merge this PR if a task for fixing this is immediately created. Or fix it before merging.

**Pull request merged**
